src/cmk/ecc_key.go

```diff
@@ -210,3 +212,62 @@ func (k *EcdsaPrivateKey) UnmarshalJSON(data []byte) error {
 	*k = EcdsaPrivateKey(pk)
 	return nil
 }
+
+//----------------------------------------------------
+// Construct key from YAML (seeding)
+//---
+func (k *EccKey) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	// Cannot use embedded 'Key' struct
+	// https://github.com/go-yaml/yaml/issues/263
+	type YamlKey struct {
+		Metadata      KeyMetadata `yaml:"Metadata"`
+		PrivateKeyPem string      `yaml:"PrivateKeyPem"`
+	}
+
+	yk := YamlKey{}
+	if err := unmarshal(&yk); err != nil {
+		return &UnmarshalYAMLError{err.Error()}
+	}
+
+	k.Type = TypeEcc
+	k.Metadata = yk.Metadata
+	pemDecoded, _ := pem.Decode([]byte(yk.PrivateKeyPem))
+	if pemDecoded == nil {
+		return &UnmarshalYAMLError{fmt.Sprintf("Unable to decode pem of key %s check the YAML.\n", k.Metadata.KeyId)}
+	}
+
+	parseResult, pkcsParseError := x509.ParseECPrivateKey(pemDecoded.Bytes)
+	if pkcsParseError != nil {
+		return &UnmarshalYAMLError{fmt.Sprintf("Unable to decode pem of key %s, Ensure it is in PKCS8 format with no password: %s.\n", k.Metadata.KeyId, pkcsParseError)}
+	}
+
+	k.PrivateKey = EcdsaPrivateKey(*parseResult)
+	var bitLen = parseResult.Curve.Params().BitSize
+
+	switch bitLen {
+	case 256:
+		k.Metadata.CustomerMasterKeySpec = SpecEccNistP256
+		k.Metadata.SigningAlgorithms = []SigningAlgorithm{SigningAlgorithmEcdsaSha256}
+	case 384:
+		k.Metadata.CustomerMasterKeySpec = SpecEccNistP384
+		k.Metadata.SigningAlgorithms = []SigningAlgorithm{SigningAlgorithmEcdsaSha384}
+	case 512:
```

Comment:
Thanks! I probably have that same bug in my java code (although we are using RSA 2048 keys since they are cheaper).  
I will fix this and also update the docs.

I tried to create a python test but didn't manage (it was an intellij thing on my old mac so didn't pursue it).

In addition I have done some work removing duplication and I will make a separate PR with that. I think it is better but does modify existing (seed) code. You can then take your pick.

If it is merged, do you have idea when it would be updated in localstack? (I realise it is a separate project but thought you could have an idea).

cheers
Mark